x/stake/keeper/slash.go

```diff
@@ -36,7 +34,12 @@ func (k Keeper) Slash(ctx sdk.Context, pubkey crypto.PubKey, infractionHeight in
 
 	validator, found := k.GetValidatorByPubKey(ctx, pubkey)
 	if !found {
-		panic(fmt.Errorf("attempted to slash a nonexistent validator with address %s", pubkey.Address()))
+		// If not found, the validator must have been overslashed and removed - so we don't need to do anything
+		// NOTE:  Correctness dependent on invariant that unbonding delegations / redelegations must also have been completely
+		//        slashed in this case - which we don't explicitly check, but should be true.
+		// Log the slash attempt for future reference (maybe we should tag it too)
+		logger.Info(fmt.Sprintf("Ignored attempt to slash a nonexistent validator with address %s", pubkey.Address()))
```

Comment:
should this not be ALL CAPS super scary text because safety _could maybe_ get broken when this happens? In production I'll be a bit scared whenever this happens 4sure

Reply:
Turned into `logger.Error`, added "WARNING" and a suggestion to "investigate immediately"